src/Cool/CoolRequest.php

```diff
@@ -25,12 +25,25 @@ function getDiscovery($server) {
     }
     $disable_checks = (bool)$default_config->get('cool')['disable_cert_check'];
 
-    $stream_context = stream_context_create([
-        'ssl' => [
-            'verify_peer'       => !$disable_checks,
-            'verify_peer_name'  => !$disable_checks,
-        ]]);
-    $res = file_get_contents($discovery_url, false, $stream_context);
+    // Previously, file_get_contents() was used to fetch the discovery xml data.
+    // Depending on the environment, it can happen that file_get_contents() will
+    // hang at the end of a stream, expecting more data.
+    // With curl, this does not happen.
+    // @todo Refactor this and use e.g. Guzzle http client.
+    $curl = \curl_init($discovery_url);
+    \curl_setopt_array($curl, [
+        \CURLOPT_RETURNTRANSFER => TRUE,
+        // Previously, when this request was done with file_get_contents() and
+        // stream_context_create(), the 'verify_peer' and 'verify_peer_name'
+        // options were set.
+        // @todo Check if an equivalent to 'verify_peer_name' exists for curl.
+        \CURLOPT_SSL_VERIFYPEER => !$disable_checks,
+    ]);
+    $res = \curl_exec($curl);
+
+    if (!$res) {
+        \Drupal::messenger()->addError(t('Cannot fetch from @url.', ['@url' => $discovery_url]));
+    }
```

Comment:
This actually happened to me all the time and slowed things down.
Not sure if we can safely include it in this PR.

Reply:
The question is which version of Collabora Online are you using. I'm currently tracking down an issue with a recent change that the Drupal module seem to trigger. And maybe it is related.

Reply:
Whichever version we get from using the 'collabora/code' docker image.

Reply:
indeed it's recent enough to have the issue.

If I apply this patch alone with the two extra lines for the SSL verification, it actually solve the problem in https://github.com/CollaboraOnline/online/issues/10042

Reply:
I will isolate the patch from this PR, and make a separate PR.
The tests no (longer) depend on the connection to collabora, so it does not impact them.

Reply:
Removed from the PR for now.